### PR TITLE
Code refactoring to make the code easier to read

### PR DESCRIPTION
**Pull request by lenvm**
### What
This PR splits `Software.ino` into a number of different files.

It also updates:
- `DUAL_CAN` to `CAN_ADDON`
- `CAN_FD` to `CANFD_ADDON`

### Why
- The code is split into different files to make it easier to read the code, and to place functions that belong together, together :)
- The updates in the defines are done, to make the code easier to read.

### How
Different folders are created, each containing a specific subset of the code:
- `can`: for functions related to CAN
- `contactorcontrol`: for functions related to contactor control
- `equipmentstopbutton`: for functions related to the equipment stop button
- `nvm`: for functions related to setting storage
- `rs485`: for functions related to RS485 / ModBus
- `seriallink`: for functions related to Serial Link

In case you have any comments or requests, please let me know!

### CODE REVIEW

File: Software/USER_SETTINGS.h [hunk 1]
```diff
@@ -64,17 +64,17 @@
...
-      OSC_40MHz  //CAN_FD option, what is your MCP2518 add-on boards crystal frequency? (Default OSC_40MHz)
-#endif
+//#define DEBUG_VIA_USB          //Enable this line to have the USB port output serial diagnostic data while program runs (WARNING, raises CPU load, do not use for production)
+//#define DEBUG_CAN_DATA         //Enable this line to print incoming/outgoing CAN & CAN-FD messages to USB serial (WARNING, raises CPU load, do not use for production)
+//#define INTERLOCK_REQUIRED     //Nissan LEAF specific setting, if enabled requires both high voltage conenctors to be seated before starting
+//#define CAN_ADDON              //Enable this line to activate an isolated secondary CAN Bus using add-on MCP2515 chip (Needed for some inverters / double battery)
+#define CRYSTAL_FREQUENCY_MHZ 8  //CAN_ADDON option, what is your MCP2515 add-on boards crystal frequency?
+//#define CANFD_ADDON           //Enable this line to activate an isolated secondary CAN-FD bus using add-on MCP2518FD chip / Native CANFD on Stark board
+#ifdef CANFD_ADDON  // CANFD_ADDON additional options if enabled
+#define CANFD_ADDON_CRYSTAL_FREQUENCY_MHZ \
```
Comment (dalathegreat) [inline Software/USER_SETTINGS.h:74]: The CANFD_ADDON_CRYSTAL_FREQUENCY_MHZ could always be defined, irregardless of CANFD_ADDON setting (makes this file smaller and easier to manage

**Comment by dalathegreat** [comment]
Very nice refactor, the silly large Software.ino refactor has been on my mind for a while, but been so busy with new integrations :sweat_smile:

File: Software/USER_SETTINGS.h (hunk 1 above)
Comment (lenvm) [reply Software/USER_SETTINGS.h:74]: Implementing this change is fine by me, to make an upgrade to the code. In this PR I have tried to just refactor the code, without introducing changes to how the code works.

Shall we keep the suggested change for another PR?

**Comment by dalathegreat** [comment]
@lenvm I think we can start looking into merging this, I just applied the critical PWM fix to main, and tagged 7.9.1 (so sorry, you'll have to deal with a minor merge conflict)

**Comment by lenvm** [comment]
@dalathegreat, no worries, I have rebased and resolved the merge conflict.

Please let me know if we need to merge #673 too, before we merge this one.

**Comment by mvgalen** [comment]
> @dalathegreat, no worries, I have rebased and resolved the merge conflict.
> 
> Please let me know if we need to merge #673 too, before we merge this one.

I have merged #673, but I do not see your rebase and the conflict is unsolved. If you commit the rebase, I will review and approve. Better not to keep a large PR like this around too long.

**Comment by dalathegreat** [comment]
@lenvm , I also merged #647 , I think we can hold off any other PRs now until this one is merged!

**Comment by lenvm** [comment]
@dalathegreat, @mvgalen, I have rebased on main again. Please review the changes now.

**Review by dalathegreat — APPROVED** [review]
LGTM :+1: Well needed refactoring! :raised_hands:

**Pull request merged**